contracts/MemeFactory.sol

```diff
@@ -204,12 +295,16 @@ abstract contract MemeFactory {
         // Zero the allocation
```

Comment:
Is there any condition where line 292 could lead to rounding errors leading to wrong distribution of allocations?